### PR TITLE
Grism mode updates for extraction

### PR DESCRIPTION
**Pull request by sosey**
Still testing this out but wanted to run it through travis. The updates generally include:
- some added logging information that's useful
- logic to correct the dispersion directions, though I haven't decided if this is correct yet, still testing
- rejection of completely off-detector orders and marking of partial orders
- rounding for the order bounding boxes to int
- addition of the grism_objects parameter optional parameter to the extract_2d step

### CODE REVIEW

File: jwst/extract_2d/grisms.py [hunk 1]
```diff
@@ -83,33 +88,57 @@ def extract_grism_objects(input_model, grism_objects=[], reference_files={}):
...
 
 
             new_model = datamodels.ImageModel(data=ext_data, err=ext_err, dq=ext_dq)
             new_model.meta.wcs = subwcs
-            util.update_s_region(new_model)
+            # Not sure this makes sense for grism exposures since the trace
+            # doesn't really have a footprint itself, it relates back to the
+            # size of the object in the direct image. So what is really wanted
+            # here?
+            # util.update_s_region(new_model)
```
Comment (nden) [inline jwst/extract_2d/grisms.py]: :+1:  on commenting this out until 7.2.

**Comment by sosey** [comment]
I think  Spec2Pipeline with nircam test data, niriss doesn't run because of missing reference files (team needs to deliver):
```
In [2]: Spec2Pipeline.call('nircam_grism_asn.json')
2017-11-03 19:35:37,575 - stpipe.Spec2Pipeline - INFO - Spec2Pipeline instance created.
2017-11-03 19:35:37,576 - stpipe.Spec2Pipeline.bkg_subtract - INFO - BackgroundStep instance created.
2017-11-03 19:35:37,577 - stpipe.Spec2Pipeline.assign_wcs - INFO - AssignWcsStep instance created.
2017-11-03 19:35:37,577 - stpipe.Spec2Pipeline.imprint_subtract - INFO - ImprintStep instance created.
2017-11-03 19:35:37,578 - stpipe.Spec2Pipeline.msa_flagging - INFO - MSAFlagOpenStep instance created.
2017-11-03 19:35:37,579 - stpipe.Spec2Pipeline.extract_2d - INFO - Extract2dStep instance created.
2017-11-03 19:35:37,579 - stpipe.Spec2Pipeline.flat_field - INFO - FlatFieldStep instance created.
2017-11-03 19:35:37,580 - stpipe.Spec2Pipeline.srctype - INFO - SourceTypeStep instance created.
2017-11-03 19:35:37,580 - stpipe.Spec2Pipeline.straylight - INFO - StraylightStep instance created.
2017-11-03 19:35:37,581 - stpipe.Spec2Pipeline.fringe - INFO - FringeStep instance created.
2017-11-03 19:35:37,582 - stpipe.Spec2Pipeline.pathloss - INFO - PathLossStep instance created.
2017-11-03 19:35:37,582 - stpipe.Spec2Pipeline.barshadow - INFO - BarShadowStep instance created.
2017-11-03 19:35:37,583 - stpipe.Spec2Pipeline.photom - INFO - PhotomStep instance created.
2017-11-03 19:35:37,583 - stpipe.Spec2Pipeline.resample_spec - INFO - ResampleSpecStep instance created.
2017-11-03 19:35:37,585 - stpipe.Spec2Pipeline.cube_build - INFO - CubeBuildStep instance created.
2017-11-03 19:35:37,585 - stpipe.Spec2Pipeline.extract_1d - INFO - Extract1dStep instance created.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for AREA reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for BARSHADOW reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for CAMERA reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for COLLIMATOR reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for CUBEPAR reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for DFLAT reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for DISPERSER reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for DISTORTION reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_distortion_0059.asdf'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for EXTRACT1D reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for FFLAT reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for FILTEROFFSET reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for FLAT reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_flat_0142.fits'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for FORE reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for FPA reference file is 'N/A'.
2017-11-03 19:35:38,190 - stpipe.Spec2Pipeline - INFO - Prefetch for FRINGE reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for IFUFORE reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for IFUPOST reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for IFUSLICER reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for MSA reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for MSAOPER reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for OTE reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for PATHLOSS reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for PHOTOM reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_photom_0050.fits'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for REGIONS reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for RESOL reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for SFLAT reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for SPECWCS reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_specwcs_0010.asdf'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for STRAYMASK reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for V2V3 reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for WAVECORR reference file is 'N/A'.
2017-11-03 19:35:38,191 - stpipe.Spec2Pipeline - INFO - Prefetch for WAVELENGTHRANGE reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_wavelengthrange_0001.asdf'.
2017-11-03 19:35:38,360 - stpipe.Spec2Pipeline - INFO - First argument nircam_grism_asn.json does not appear to be a model
2017-11-03 19:35:38,396 - stpipe.Spec2Pipeline - INFO - Step Spec2Pipeline running with args ('nircam_grism_asn.json',).
2017-11-03 19:35:38,397 - stpipe.Spec2Pipeline - INFO - Starting calwebb_spec2 ...
2017-11-03 19:35:38,402 - stpipe.Spec2Pipeline - INFO - Processing product test_disperse_f335m
2017-11-03 19:35:38,402 - stpipe.Spec2Pipeline - INFO - Working on input dispersed_ramp_from_test.fits ...
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for CAMERA reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for COLLIMATOR reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for DISPERSER reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for DISTORTION reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_distortion_0059.asdf'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for FILTEROFFSET reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for FORE reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for FPA reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for IFUFORE reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for IFUPOST reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for IFUSLICER reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for MSA reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for OTE reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for REGIONS reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for SPECWCS reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_specwcs_0010.asdf'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for V2V3 reference file is 'N/A'.
2017-11-03 19:35:38,632 - stpipe.Spec2Pipeline.assign_wcs - INFO - Prefetch for WAVELENGTHRANGE reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_wavelengthrange_0001.asdf'.
2017-11-03 19:35:38,666 - stpipe.Spec2Pipeline.assign_wcs - INFO - Step assign_wcs running with args (<jwst.datamodels.image.ImageModel object at 0x11a86b6a0>,).
2017-11-03 19:35:39,576 - stpipe.Spec2Pipeline.assign_wcs - INFO - assign_wcs did not update S_REGION for type NRC_GRISM
2017-11-03 19:35:39,577 - stpipe.Spec2Pipeline.assign_wcs - INFO - COMPLETED assign_wcs
2017-11-03 19:35:39,630 - stpipe.Spec2Pipeline.assign_wcs - INFO - Step assign_wcs done
2017-11-03 19:35:39,678 - stpipe.Spec2Pipeline.bkg_subtract - INFO - Step bkg_subtract running with args (<jwst.datamodels.image.ImageModel object at 0x11263a080>, []).
2017-11-03 19:35:39,742 - stpipe.Spec2Pipeline.bkg_subtract - INFO - No 'wfssbkg' reference file available; skipping background subtraction.
2017-11-03 19:35:40,075 - stpipe.Spec2Pipeline.bkg_subtract - INFO - Step bkg_subtract done
2017-11-03 19:35:40,235 - stpipe.Spec2Pipeline.flat_field - INFO - Prefetch for DFLAT reference file is 'N/A'.
2017-11-03 19:35:40,235 - stpipe.Spec2Pipeline.flat_field - INFO - Prefetch for FFLAT reference file is 'N/A'.
2017-11-03 19:35:40,235 - stpipe.Spec2Pipeline.flat_field - INFO - Prefetch for FLAT reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_flat_0142.fits'.
2017-11-03 19:35:40,235 - stpipe.Spec2Pipeline.flat_field - INFO - Prefetch for SFLAT reference file is 'N/A'.
2017-11-03 19:35:40,285 - stpipe.Spec2Pipeline.flat_field - INFO - Step flat_field running with args (<jwst.datamodels.image.ImageModel object at 0x112690ac8>,).
2017-11-03 19:35:41,110 - stpipe.Spec2Pipeline.flat_field - INFO - Step flat_field done
2017-11-03 19:35:41,261 - stpipe.Spec2Pipeline.extract_2d - INFO - Prefetch for WAVECORR reference file is 'N/A'.
2017-11-03 19:35:41,261 - stpipe.Spec2Pipeline.extract_2d - INFO - Prefetch for WAVELENGTHRANGE reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_wavelengthrange_0001.asdf'.
2017-11-03 19:35:41,299 - stpipe.Spec2Pipeline.extract_2d - INFO - Step extract_2d running with args (<jwst.datamodels.image.ImageModel object at 0x11261b898>,).
2017-11-03 19:35:41,426 - stpipe.Spec2Pipeline.extract_2d - INFO - EXP_TYPE is NRC_GRISM
2017-11-03 19:35:41,770 - stpipe.Spec2Pipeline.extract_2d - INFO - Partial order on detector for obj: 10 order: 1
2017-11-03 19:35:41,774 - stpipe.Spec2Pipeline.extract_2d - INFO - Partial order on detector for obj: 10 order: 2
2017-11-03 19:35:41,774 - stpipe.Spec2Pipeline.extract_2d - INFO - Grism object list created from source catalog: test_disperse_f335m_rate_updated_bboxradecskybox_short.ecsv
2017-11-03 19:35:41,774 - stpipe.Spec2Pipeline.extract_2d - INFO - Creating output model
2017-11-03 19:35:42,640 - stpipe.Spec2Pipeline.extract_2d - INFO - Subarray extracted for obj: 10 order: 1:
2017-11-03 19:35:42,640 - stpipe.Spec2Pipeline.extract_2d - INFO - Subarray extents are: (xmin:0, ymin:441), (xmax:584, ymax:521)
2017-11-03 19:40:02,434 - stpipe.Spec2Pipeline.extract_2d - INFO - Subarray extracted for obj: 10 order: 2:
2017-11-03 19:40:02,434 - stpipe.Spec2Pipeline.extract_2d - INFO - Subarray extents are: (xmin:1613, ymin:441), (xmax:2048, ymax:521)
2017-11-03 19:53:07,317 - stpipe.Spec2Pipeline.extract_2d - INFO - Step extract_2d done
2017-11-03 19:53:07,414 - stpipe.Spec2Pipeline.srctype - INFO - Step srctype running with args (<jwst.datamodels.multislit.MultiSlitModel object at 0x11255a0b8>,).
2017-11-03 19:53:07,427 - stpipe.Spec2Pipeline.srctype - INFO - Input EXP_TYPE is NRC_GRISM
2017-11-03 19:53:07,427 - stpipe.Spec2Pipeline.srctype - WARNING - EXP_TYPE NRC_GRISM not applicable to this operation
2017-11-03 19:53:07,427 - stpipe.Spec2Pipeline.srctype - WARNING - Step will be skipped
2017-11-03 19:53:07,431 - stpipe.Spec2Pipeline.srctype - INFO - Step srctype done
2017-11-03 19:53:07,550 - stpipe.Spec2Pipeline.photom - INFO - Prefetch for AREA reference file is 'N/A'.
2017-11-03 19:53:07,550 - stpipe.Spec2Pipeline.photom - INFO - Prefetch for PHOTOM reference file is '/Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_photom_0050.fits'.
2017-11-03 19:53:07,597 - stpipe.Spec2Pipeline.photom - INFO - Step photom running with args (<jwst.datamodels.multislit.MultiSlitModel object at 0x118e39470>,).
2017-11-03 19:53:07,660 - stpipe.Spec2Pipeline.photom - INFO - Using photom reference file: /Users/sosey/crds_cache/references/jwst/nircam/jwst_nircam_photom_0050.fits
2017-11-03 19:53:07,712 - stpipe.Spec2Pipeline.photom - INFO - Using area reference file: N/A
2017-11-03 19:53:07,712 - stpipe.Spec2Pipeline.photom - INFO - Using instrument: NIRCAM
2017-11-03 19:53:07,712 - stpipe.Spec2Pipeline.photom - INFO -  detector: NRCALONG
2017-11-03 19:53:07,712 - stpipe.Spec2Pipeline.photom - INFO -  exp_type: NRC_GRISM
2017-11-03 19:53:07,713 - stpipe.Spec2Pipeline.photom - INFO -  filter: F360M
2017-11-03 19:53:07,713 - stpipe.Spec2Pipeline.photom - INFO -  pupil: GRISMR
2017-11-03 19:53:07,750 - stpipe.Spec2Pipeline.photom - WARNING - Did not find a match in the ref file, so returning conversion factor 0.0
2017-11-03 19:53:07,750 - stpipe.Spec2Pipeline.photom - INFO - Writing PHOTMJSR with value: 0
2017-11-03 19:53:07,757 - stpipe.Spec2Pipeline.photom - INFO - Step photom done
2017-11-03 19:53:07,873 - stpipe.Spec2Pipeline.extract_1d - INFO - Prefetch for EXTRACT1D reference file is 'N/A'.
2017-11-03 19:53:07,937 - stpipe.Spec2Pipeline.extract_1d - INFO - Step extract_1d running with args (<jwst.datamodels.multislit.MultiSlitModel object at 0x118ecf9e8>,).
2017-11-03 19:53:07,950 - stpipe.Spec2Pipeline.extract_1d - INFO - No EXTRACT1D reference file will be used
2017-11-03 19:53:07,971 - stpipe.Spec2Pipeline.extract_1d - INFO - Working on slit 10
2017-11-03 19:53:49,434 - stpipe.Spec2Pipeline.extract_1d - WARNING - wcs.bounding_box is None
```
it's taking a while to complete locally from the spec2 call, but when I started x1d up with the previous steps output it completed reasonably:
```
2017-11-03 20:55:46,608 - stpipe.Extract1dStep - WARNING - No relsens for current slit, so can't compute flux.
2017-11-03 20:55:46,900 - stpipe.Extract1dStep - INFO - Working on slit 10
2017-11-03 20:55:46,913 - stpipe.Extract1dStep - WARNING - wcs.bounding_box is None
2017-11-03 20:56:12,468 - stpipe.Extract1dStep - WARNING - No relsens for current slit, so can't compute flux.
2017-11-03 20:56:12,831 - stpipe.Extract1dStep - INFO - Step Extract1dStep done
```
also, james' updates looked good.

@hbushouse  I'll let you merge this ?

**Comment by hbushouse** [comment]
Which NIRISS reference files are missing? Is the team (either NIRISS instrument or WFSS working group) aware that we're expecting them to create and deliver them?

**Comment by sosey** [comment]
we've been waiting for mike wolfe to redeliver the distortion reference files that will match with the wfss modes.  I'm also trying to track this down for extract1d:
```
5]: a=step('niriss_grism_asn.json')
CRDS - ERROR -  Error determining best reference for 'extract1d'  =    parameter='META.EXPOSURE.TYPE' value='NIS_WFSS' is not in ['NIS_SOSS']
2017-11-04 10:30:20,643 - stpipe.Spec2Pipeline - ERROR -  Error determining best reference for 'extract1d'  =    parameter='META.EXPOSURE.TYPE' value='NIS_WFSS' is not in ['NIS_SOSS']
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/crds_client.py in get_multiple_reference_paths(dataset_model, reference_file_types)
     73         refpaths = { reftype:_BESTREFS_CACHE[(filename, reftype)]
---> 74                      for reftype in reference_file_types }
     75         log.verbose("Using cached bestrefs for", repr(filename), "with types", repr(reference_file_types))

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/crds_client.py in <dictcomp>(.0)
     73         refpaths = { reftype:_BESTREFS_CACHE[(filename, reftype)]
---> 74                      for reftype in reference_file_types }
     75         log.verbose("Using cached bestrefs for", repr(filename), "with types", repr(reference_file_types))

KeyError: ('niriss_F200W_gr150r_position1_rate.fits', 'area')

During handling of the above exception, another exception occurred:

CrdsLookupError                           Traceback (most recent call last)
<ipython-input-15-2d99b9fc3bc0> in <module>()
----> 1 a=step('niriss_grism_asn.json')

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py in run(self, *args)
    343             if (len(args) and len(self.reference_file_types) and not self.skip
    344                 and self.prefetch_references):
--> 345                 self._precache_reference_files(args[0])
    346 
    347             self.log.info(

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py in _precache_reference_files(self, input_file)
    572         try:
    573             with datamodels.open(input_file) as model:
--> 574                 self._precache_reference_files_opened(model)
    575         except (ValueError, TypeError, IOError):
    576             self.log.info(

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py in _precache_reference_files_opened(self, model_or_container)
    593             # recurse on each contained model
    594             for contained_model in model_or_container:
--> 595                 self._precache_reference_files_opened(contained_model)
    596         else:
    597             # precache a single model object

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py in _precache_reference_files_opened(self, model_or_container)
    596         else:
    597             # precache a single model object
--> 598             self._precache_reference_files_impl(model_or_container)
    599 
    600     def _precache_reference_files_impl(self, model):

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py in _precache_reference_files_impl(self, model)
    614         fetch_types = sorted(set(self.reference_file_types) - set(ovr_refs.keys()))
    615 
--> 616         crds_refs = crds_client.get_multiple_reference_paths(model, fetch_types)
    617 
    618         ref_path_map = dict(list(crds_refs.items()) + list(ovr_refs.items()))

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/crds_client.py in get_multiple_reference_paths(dataset_model, reference_file_types)
     80 
     81         # Cache prefetch-like results
---> 82         refpaths = _get_refpaths(data_dict, tuple(reference_file_types))
     83 
     84         # Cache results for each individual reftype, as-in get_reference_file().

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/jwst-0.7.8rc6.dev73-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/crds_client.py in _get_refpaths(data_dict, reference_file_types)
    118     try:
    119         with crds_cache_locking.get_cache_lock():
--> 120             bestrefs = crds.getreferences(data_dict, reftypes=reference_file_types, observatory="jwst")
    121     except crds.CrdsBadRulesError as exc:
    122         raise crds.CrdsBadRulesError(str(exc))

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/crds/core/heavy_client.py in getreferences(parameters, reftypes, context, ignore_cache, observatory, fast)
    104     # should work without network access if files are already cached.
    105     best_refs_paths = api.cache_references(
--> 106         final_context, bestrefs, ignore_cache=ignore_cache)
    107 
    108     return best_refs_paths

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/crds/client/api.py in cache_references(pipeline_context, bestrefs, ignore_cache)
    756     Returns:   { reference_keyword :  reference_local_filepath ... }
    757     """
--> 758     wanted = _get_cache_filelist_and_report_errors(bestrefs)
    759 
    760     localrefs = FileCacher(pipeline_context, ignore_cache, raise_exceptions=False).get_local_files(wanted)[0]

~/miniconda3/envs/jwst-test/lib/python3.6/site-packages/crds/client/api.py in _get_cache_filelist_and_report_errors(bestrefs)
    793             log.error(str(last_error))
    794     if last_error is not None:
--> 795         raise last_error
    796     return wanted
    797 

CrdsLookupError: Error determining best reference for 'extract1d'  =    parameter='META.EXPOSURE.TYPE' value='NIS_WFSS' is not in ['NIS_SOSS']

```

**Comment by hbushouse** [comment]
When I tried testing a NIRCam grism image it took 10 mins or so in extract_2d, like yours, but then it just sat for hours in extract_1d without ever finishing. Had to kill it off. What's going on in there?

**Comment by nden** [comment]
The rmap for the extract_1d reference needs to be modified. It has only one entry currently - for NIS_SOSS.

**Comment by sosey** [comment]
@nden yah, I was hoping mike would be delivering this with the updated distortion files

**Comment by sosey** [comment]
@hbushouse not sure, all the steps work fine by themselves on input/output, this only happened to me with the spec2pipeline...not sure what's going on yet.

**Pull request merged**